src/udiskslinuxblock.c

```diff
@@ -3423,19 +3461,22 @@ handle_open_for_benchmark (UDisksBlock           *block,
   g_variant_lookup (options, "writable", "b", &opt_writable);
 
   if (opt_writable)
-    open_flags = O_RDWR  | O_EXCL;
+    {
+      open_flags = O_EXCL | O_DIRECT | O_SYNC | O_CLOEXEC;
+      open_mode = "rw";
+    }
   else
-    open_flags = O_RDONLY;
-
-  open_flags |= O_DIRECT | O_SYNC | O_CLOEXEC;
```

Comment:
Maybe make these the default flags and `open_flags |= O_EXCL;` if writable? Obviously only if you want to stay with this `mode` thing.